ocaml/runtime/gc_ctrl.c

```diff
@@ -299,8 +299,7 @@ CAMLprim value caml_gc_compaction(value v)
   Caml_check_caml_state();
   CAML_EV_BEGIN(EV_EXPLICIT_GC_COMPACT);
   CAMLassert (v == Val_unit);
-  value exn = gc_major_exn(1);
-  ++ Caml_state->stat_forced_major_collections;
+  value exn = gc_full_major_exn();
```

Comment:
This looks broken to me? You're no longer passing `force_compaction = 1` so I don't see how this would actually trigger a compaction.

Reply:
Hah, you are right!  Will fix.